drivers: sdhc: set 'sdhc_driver_api' as 'static const'

ndrs-pst opened the pull request:
This change marks each instance of the `sdhc_driver_api` as `static const`. The rationale is that `sdhc_driver_api` is used for declaring internal module interfaces and is not intended to be modified at runtime. By using `static const`, we ensure immutability, leading to usage of only **.rodata** and a reduction in the **.data** area.

Additionally, it's worth noting that the directory **drivers/sdhc/sdhc_cdns/** might have been added unintentionally, as CMakeLists.txt is searching for the files located outside this directory.

danieldegrasse reviewed (commented):
Thanks for catching this- it does not seem like `sdhc_cdns/sdhc_cdnc.c` is being included in the build. @roymurlidhar, can you confirm these files are ok to remove?

Besides this, a few broad comments:
- please split this PR into two commits:
   - commit adding `const` specifier to API definitions
   - commit removing sdhc_cdns files

ndrs-pst committed:
drivers: sdhc: set 'sdhc_driver_api' as 'static const'

This change marks each instance of the 'api' as 'static const'.
The rationale is that 'api' is used for declaring internal
module interfaces and is not intended to be modified at runtime.
By using 'static const', we ensure immutability, leading to usage of only
.rodata and a reduction in the .data area.

Signed-off-by: Pisit Sawangvonganan <pisit@ndrsolution.com>

ndrs-pst commented:
@danieldegrasse Already split the PR into two separate commits, thanks.

ndrs-pst commented:
@danieldegrasse, since this PR https://github.com/zephyrproject-rtos/zephyr/pull/67727 already addressed https://github.com/zephyrproject-rtos/zephyr/pull/66996/commits/0626d7fe03f450146a28e3dd354744d4a5101014 so I just update to remain just one commit.
